### DISCUSSION TITLE
Unique keys on update

### DISCUSSION THREAD

**Discussion by BananaMasterz** [simolus3/drift#2145] [Q&A]
When I try to do `update(table)..where.......write(stuff)` on a table that has

```dart
@override
List<Set<Column>> get uniqueKeys => [
       {column1, column2}
];
```

I get `UNIQUE constraint failed` am I doing something wrong or do unique keys work differently that that?

**Comment by simolus3** [comment]
You'll get this error if the update causes multiple rows to have the same values for the column. This could happen if the `where` matches more than one row, or if you're updating one row to the values another row already has. It's hard to say what exactly is causing the problem, but you can probably find out more by running a select with the same `where` or by filtering for the values you're trying to update to see if another row could cause the conflict.

**Reply by BananaMasterz** [reply]
The thing is, the table has only one row and I'm trying to update that one row (referring to it by ID specifically)... Ok if nothing obviously wrong with what I'm doing I'll try checking more on what's going on. Thanks!
